### ISSUE TITLE
Patch for `__pwInitScripts`

### ISSUE THREAD

**Issue by tonybruess** [rebrowser/rebrowser-patches#52]
Creating this issue so I can track it 😄 

Need to fix the `__pwInitScripts` detection introduced in playwright 1.46.1

**Comment by ofirk** [comment]
Same issue in my side as well, can it cause a detection?
Thanks again for a great repo!

**Comment by daniellionel01** [comment]
I got it fixed with the following code:

```typescript
await browser.addInitScript(() => {
  delete (window as any).__pwInitScripts;
})
```

**Comment by gohoski** [comment]
> I got it fixed with the following code:
> 
> ```ts
> await browser.addInitScript(() => {
>   delete (window as any).__pwInitScripts;
> })
> ```

Works only every second try. Is there any solution that 100% works?

**Comment by gohoski** [comment]
Solution via a proxy
```typescript
    await context.addInitScript(() => {
      // Delete the property if it exists
      delete (window as any).__pwInitScripts;

      // Redefine `Object.getOwnPropertyNames`
      const originalGetOwnPropertyNames = Object.getOwnPropertyNames;
      Object.getOwnPropertyNames = function (obj) {
        const props = originalGetOwnPropertyNames(obj);
        return props.filter((prop) => prop !== '__pwInitScripts');
      };

      // Use a Proxy to handle access to `window`
      const windowHandler = {
        get(target: any, prop: string) {
          if (prop === '__pwInitScripts') {
            return undefined; // Hide the property
          }
          return Reflect.get(target, prop);
        },
        has(target: any, prop: string) {
          if (prop === '__pwInitScripts') {
            return false; // Prevent detection via "in" operator
          }
          return Reflect.has(target, prop);
        },
      };

      const proxiedWindow = new Proxy(window, windowHandler);
      Object.defineProperty(globalThis, 'window', {
        value: proxiedWindow,
        configurable: false,
        writable: false,
      });
    })
```

**Comment by daniellionel01** [comment]
I'd be careful with using proxies. The presence of a proxy can be detected by client bot detector libraries and a proxy on the window object is probably going to ring some alarm bells. Just FYI.

**Comment by DeHby** [comment]
Rebrowser-patches doesn't seem to sync this patch? I can only manually modify the lib code in node_module

**Comment by daniellionel01** [comment]
> Rebrowser-patches doesn't seem to sync this patch? I can only manually modify the lib code in node_module

I wouldn't recommend rebrowser-patches tbh, I've found a lot of success with https://github.com/Kaliiiiiiiiii-Vinyzu/patchright-nodejs

**Comment by DeHby** [comment]
> > Rebrowser-patches doesn't seem to sync this patch? I can only manually modify the lib code in node_module
> 
> I wouldn't recommend rebrowser-patches tbh, I've found a lot of success with https://github.com/Kaliiiiiiiiii-Vinyzu/patchright-nodejs

GitHub has too many various magic modification frameworks,I don't know what's best😂

**Comment by daniellionel01** [comment]
> > > Rebrowser-patches doesn't seem to sync this patch? I can only manually modify the lib code in node_module
> > 
> > 
> > I wouldn't recommend rebrowser-patches tbh, I've found a lot of success with https://github.com/Kaliiiiiiiiii-Vinyzu/patchright-nodejs
> 
> GitHub has too many various magic modification frameworks,I don't know what's best😂

Yeah it's quite overwhelming tbh. We've done a huge analysis at work for an automation product, and patchright yielded the best results for us
